addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/handler/solarviewHandler.java

```diff
@@ -0,0 +1,417 @@
+                    case CHANNEL_PAC:
+                    case CHANNEL_UDC:
+                    case CHANNEL_IDC:
+                    case CHANNEL_UDCB:
+                    case CHANNEL_IDCB:
+                    case CHANNEL_UDCC:
+                    case CHANNEL_IDCC:
+                    case CHANNEL_UL1:
+                    case CHANNEL_IL1:
+                    case CHANNEL_TKK:
+                        updateState(channelUID, getChannelValue(channelUID.getId()));
+                        break;
+                    default:
+                        logger.error("Command received for an unknown channel: {}", channelUID.getId());
+                        break;
+                }
+            }
+        } else {
+            logger.error("Command {} is not supported for channel: {}", command, channelUID.getId());
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+        Collection<ConfigStatusMessage> configStatus = new ArrayList<>();
+        logger.trace("getConfigStatus() called.");
+        return configStatus;
+    }
+
+    /**
+     * Ask the Solarview server for an information update of the
+     * set of information (identified by {@link queryString}) and
+     * store the results within a local storage where the information
+     * items could be retrieved by {@link getChannelValue}.
+     *
+     * @return boolean success of the information request.
+     */
+    private synchronized boolean updateEnergyDataFromServer() {
+        logger.trace("updateEnergyDataFromServer() called.");
+        final Socket clientSocket;
+        try {
+            clientSocket = new Socket(hostname, SERVER_CONNECTION_DSTPORT);
+        } catch (Exception e) {
+            logger.info("Exception occurred on socket creation: {}", e.getMessage());
+            energyInformation = null;
+            logger.debug("updateEnergyDataFromServer() return false.");
+            return (false);
+        }
+
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    DataOutputStream outToServer = new DataOutputStream(clientSocket.getOutputStream());
+                    BufferedReader inFromServer = new BufferedReader(
+                            new InputStreamReader(clientSocket.getInputStream()));
+                    logger.trace("updateEnergyDataFromServer() using query string '{}'.", queryString);
+                    outToServer.writeBytes(queryString + SOLARVIEW_REQUEST_EOL);
+                    energyInformation = inFromServer.readLine();
+                    logger.trace("updateEnergyDataFromServer() got response.");
+
+                } catch (IOException e) {
+                    logger.info("Exception occurred during reading on socket: {}", e.getMessage());
+                    energyInformation = null;
+                }
+            }
+        }).start();
+
+        try {
```

Comment:
What are you doing here? Are you starting a background thread but wait until it returns to process it in the main thread, this is not how it should be done. If it takes longer than 500ms to do the update please do it all in a background thread